From fa54b7252999e05dff732242c1d81e1a14a61d1f Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Tue, 1 Nov 2016 15:48:15 +0100
Subject: Revert "wined3d: Let buffer_direct_upload() invalidate
 STATE_INDEXBUFFER if needed."

This reverts commit b1cc82a29f50dd7cc6703c33ee11df0f0db84b02.
---
 dlls/wined3d/buffer.c | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/dlls/wined3d/buffer.c b/dlls/wined3d/buffer.c
index 8943144..b50132a 100644
--- a/dlls/wined3d/buffer.c
+++ b/dlls/wined3d/buffer.c
@@ -668,13 +668,15 @@ drop_query:
 }
 
 /* The caller provides a GL context */
-static void buffer_direct_upload(struct wined3d_buffer *This, struct wined3d_context *context, DWORD flags)
+static void buffer_direct_upload(struct wined3d_buffer *This, const struct wined3d_gl_info *gl_info, DWORD flags)
 {
-    const struct wined3d_gl_info *gl_info = context->gl_info;
-    unsigned int start, len;
     BYTE *map;
+    UINT start, len;
 
-    buffer_bind(This, context);
+    /* This potentially invalidates the element array buffer binding, but the
+     * caller always takes care of this. */
+    GL_EXTCALL(glBindBuffer(This->buffer_type_hint, This->buffer_object));
+    checkGLcall("glBindBuffer");
     if (gl_info->supported[ARB_MAP_BUFFER_RANGE])
     {
         GLbitfield mapflags;
@@ -684,7 +686,7 @@ static void buffer_direct_upload(struct wined3d_buffer *This, struct wined3d_con
         else if (!(flags & WINED3D_BUFFER_SYNC))
             mapflags |= GL_MAP_UNSYNCHRONIZED_BIT;
         map = GL_EXTCALL(glMapBufferRange(This->buffer_type_hint, 0,
-                This->resource.size, mapflags));
+                    This->resource.size, mapflags));
         checkGLcall("glMapBufferRange");
     }
     else
@@ -865,9 +867,12 @@ void buffer_internal_preload(struct wined3d_buffer *buffer, struct wined3d_conte
 
         /* Nothing to do because we locked directly into the vbo */
         if (!(buffer->flags & WINED3D_BUFFER_DOUBLEBUFFER))
+        {
             return;
+        }
+
+        buffer_direct_upload(buffer, gl_info, flags);
 
-        buffer_direct_upload(buffer, context, flags);
         return;
     }
 
-- 
2.9.0

